packages/puppeteer-core/src/util/Deferred.ts

```diff
@@ -37,26 +75,24 @@ export class Deferred<T, V extends Error = Error> {
     }
   }
 
-  #finish(value: T | V | TimeoutError) {
-    clearTimeout(this.#timeoutId);
-    this.#value = value;
-    this.#resolver();
-  }
-
   resolve(value: T): void {
     if (this.#isRejected || this.#isResolved) {
       return;
     }
     this.#isResolved = true;
-    this.#finish(value);
+    clearTimeout(this.#timeoutId);
+    this.#value = value;
+    this.#resolve(value);
   }
 
   reject(error: V | TimeoutError): void {
     if (this.#isRejected || this.#isResolved) {
       return;
     }
     this.#isRejected = true;
-    this.#finish(error);
+    clearTimeout(this.#timeoutId);
+    this.#value = error;
+    this.#reject(error);
```

Comment:
There is a reason why `Deferred` was changed not use reject. Please rereview the PR that made the change or lets chat offline.

Reply:
Ah, yes, I remember.